ext/liquid_c/vm.c

```diff
@@ -361,7 +363,7 @@ static VALUE vm_render_until_error(VALUE uncast_args)
             }
 
             case OP_WRITE_NODE:
-                rb_funcall(cLiquidBlockBody, id_render_node, 3, vm->context.self, output, (VALUE)*const_ptr++);
+                rb_funcall(cLiquidBlockBody, id_render_node, 3, vm->context.self, output, *tags_ptr++);
```

Comment:
Yeah I don't think the current way is the best way of implementing. I didn't store tags inside of the constant table because it added complexity (the need to iterate over the constants array and remove all the tags, and then during deserialization update the correct indexes in the constants array). I think in the future, we be storing indexes for constants/tags which would eliminate the need for both the `const_ptr` and the `tags_ptr`.